app/controllers/api/base_controller/renderer.rb

```diff
@@ -166,7 +166,9 @@ def virtual_attribute_search(resource, attribute)
         if resource.class < ApplicationRecord
           # is relation in 'attribute' variable plural in the model class (from 'resource.class') ?
           if [:has_many, :has_and_belongs_to_many].include?(resource.class.reflection_with_virtual(attribute).try(:macro))
-            Rbac.filtered(resource.public_send(attribute))
+            resource_attr = resource.public_send(attribute)
+            return resource_attr unless resource_attr.first.kind_of?(ApplicationRecord)
```

Comment:
I'm having a hard time trying to parse this mentally - perhaps a comment would help here? If I understand this correctly (I may not) - I'm concerned whether `Model#some_attribute` is guaranteed to respond to `first`?